command/auto_complete.go

```diff
@@ -0,0 +1,246 @@
+		var err error
+		u, err = url.New(u.Absolute() + "/")
+		if err != nil {
+			return
+		}
+	}
+
+	i := 0
+	for obj := range (*client).List(ctx, u, false) {
+		if i > count {
+			break
+		}
+		if obj.Err != nil {
+			return
+		}
+		fmt.Println(formatSuggestionForShell(shell, obj.URL.Absolute(), argToBeCompleted))
+		i++
+	}
+}
+
+func printAutocompletionInstructions(shell string) {
+	var script string
+	baseShell := filepath.Base(shell)
+	instructions := `# To enable autocompletion you should add the following script to startup scripts of your shell.
+# It is probably located at ~/.` + baseShell + "rc"
+
+	switch baseShell {
+	case "zsh":
+		script = zsh
+	case "bash":
+		script = bash
+	case "pwsh":
+		script = pwsh
+		instructions = `# To enable autocompletion you should save the following script to a file named "s5cmd.ps1" and execute it.
+# To persist it you should add the path of "s5cmd.ps1" file to profile file (which you can locate with $profile) to automatically execute "s5cmd.ps1" on every shell start up.`
+	default:
+		instructions = `# We couldn't recognize your SHELL "` + baseShell + `".
+# Shell completion is supported only for bash, pwsh and zsh.
+# Make sure that your SHELL environment variable is set accurately.`
+	}
+
+	fmt.Println(instructions)
+	fmt.Println(script)
+}
+
+func formatSuggestionForShell(baseShell, suggestion, argToBeCompleted string) string {
+	switch baseShell {
+	case "bash":
+		var prefix string
+		suggestions := make([]string, 0, 2)
+		if i := strings.LastIndex(argToBeCompleted, ":"); i >= 0 && baseShell == "bash" {
+			// include the original suggestion in case that COMP_WORDBREAKS does not contain :
+			// or that the argToBeCompleted was quoted.
+			// Bash doesn't split on : when argument is quoted even if : is in COMP_WORDBREAKS
+			suggestions = append(suggestions, suggestion)
+			prefix = argToBeCompleted[0 : i+1]
+		}
+		suggestions = append(suggestions, strings.TrimPrefix(suggestion, prefix))
+		return strings.Join(suggestions, "\n")
+	case "zsh":
+		// replace every colon : with \:	if shell is zsh
+		// colons are used as a seperator for the autocompletion script
+		// so "literal colons in completion must be quoted with a backslash"
+		// see also https://zsh.sourceforge.io/Doc/Release/Completion-System.html#:~:text=This%20is%20followed,as%20name1%3B
+		return escapeColon(suggestion)
+	default:
+		return suggestion
+	}
+}
+
+// replace every colon : with \:
+func escapeColon(str ...interface{}) string {
+	return strings.ReplaceAll(fmt.Sprint(str...), ":", `\:`)
+}
+
+func parseArgumentToComplete(ctx *cli.Context) string {
+	var arg string
+	args := ctx.Args()
+	l := args.Len()
+
+	if l > 0 {
+		arg = args.Get(l - 1)
+	}
+
+	// argument may start with a quotation mark, in this case we want to trim that before
+	// checking if it has prefix s3://
+	// Beware that we only want to trim the first char, not all of the leading
+	// quotation marks, because those quotation marks may be actual charactes.
```

Comment:
nitpick/typo: let's limit length of the comments to ~80 chars per lone.
```suggestion
	// argument may start with a quotation mark, in this case we want to trim
	// that before checking if it has prefix 's3://'.
	// Beware that we only want to trim the first char, not all of the leading
	// quotation marks, because those quotation marks may be actual characters.
```